### PR TITLE
GET request data retrieval from request.args in Flask

### PR DESCRIPTION
**Pull request by basvandriel**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

*

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Thanks for adding the `RELEASE.md` file!



Here's a preview of the changelog:

---

GET-request's data gets no longer retrieved from request.json


---
Here's the preview release card for twitter:




Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to @bvandriel for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

**Review by patrick91 — CHANGES_REQUESTED** [review]
Hey, sorry for taking so long to review this :)

So, overall it looks great! I didn't realise that the tests where using "GET" instead of "POST", your change makes sense, since we want to allow queries via "GET", but before merging this, can we make sure we don't allow mutations via "GET"?

File: strawberry/flask/views.py [hunk 1]
```diff
@@ -42,20 +42,32 @@ def dispatch_request(self):
...
+        # Shorthand for using it
 
         try:
             request_data = parse_request_data(data)
         except MissingQueryError:
             return Response("No valid query was provided for the request", 400)
 
+        # Move to the @mutation annotation later?
+        if using_get and "mutation" in request_data.query:
+            return Response("GET requests don't support mutations", 400)
```
Comment (patrick91) [inline strawberry/flask/views.py:69]: not sure this it the best check for this, mutation might appear in a random field of the query :)

I think we might need to do something more clever here and change our `execute` and `execute_sync` functions on the schema to accept a list of operation type they can allow.

so then in dispatch_request we do something like this:


```
        result = self.schema.execute_sync(
            request_data.query,
            variable_values=request_data.variables,
            context_value=context,
            operation_name=request_data.operation_name,
            root_value=self.get_root_value(),
            allowed_operation_types=[QUERY] if get else [MUTATION, QUERY, SUBSCRIPTION]
        )
```

I'm open to other suggestions :) /cc @BryceBeagle for more ideas 😊

Comment (BryceBeagle) [reply strawberry/flask/views.py:72]: Yeah, as `request_data.query` is a `str`, I think using `in` is probably pretty unsafe. I don't have any other better solutions though

Comment (basvandriel) [reply strawberry/flask/views.py:72]: Currently made use of the `parse` function from `graphql-core` to retrieve the operation type. https://github.com/strawberry-graphql/strawberry/pull/1759/files#diff-88aa6fd17e4c6feac6e7152ebd3f2b8f972544c444a071b550e4d23061b97a3fR76

**Comment by patrick91** [comment]
This was integrated in https://github.com/strawberry-graphql/strawberry/pull/1686 I'll close

**Pull request closed without merging**